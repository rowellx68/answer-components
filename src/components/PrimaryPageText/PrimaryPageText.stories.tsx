import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import PrimaryPageText, { PrimaryPageTextProps } from './PrimaryPageText'

export default {
  title: 'Components/Text/PrimaryPageText',
  component: PrimaryPageText,
  argTypes: {
    className: { control: 'text' },
    text: { control: 'text' },
  },
} as Meta;

const Template: Story<PrimaryPageTextProps> = (args) => (
  <PrimaryPageText {...args} />
)

export const BasicUsage = Template.bind({})
BasicUsage.args = {
  className: 'pb-6',
  text: 'We are Answer Digital.'
}
