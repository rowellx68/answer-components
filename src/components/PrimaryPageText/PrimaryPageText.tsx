import React from 'react'

export interface PrimaryPageTextProps {
  className?: string
  text: string
}

const PrimaryPageText: React.FC<PrimaryPageTextProps> = ({ text, className = '' }) => {
  return (
    <h1 className={`text-5xl md:text-6xl font-black bg-gradient-to-r from-pink-600 via-blue-700 to-blue-700 bg-clip-text text-transparent ${className}`}>
      {text}
    </h1>
  )
}

export default PrimaryPageText
