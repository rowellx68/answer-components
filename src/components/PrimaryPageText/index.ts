import PrimaryPageText from './PrimaryPageText'
export default PrimaryPageText
