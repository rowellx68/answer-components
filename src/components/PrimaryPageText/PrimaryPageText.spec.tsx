import React from 'react'
import renderer from 'react-test-renderer'
import PrimaryPageText from './PrimaryPageText'

test('Renders PrimaryPageText correctly', () => {
  const element = renderer
    .create(<PrimaryPageText text="We are Answer Digital." />)
    .toJSON()

  expect(element).toMatchSnapshot()
})

test('Renders PrimaryPageText correctly when passing className', () => {
  const element = renderer
    .create(<PrimaryPageText className="pb-6" text="We are Answer Digital." />)
    .toJSON()

  expect(element).toMatchSnapshot()
})
