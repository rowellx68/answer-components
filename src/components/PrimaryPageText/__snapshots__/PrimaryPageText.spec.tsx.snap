// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders PrimaryPageText correctly 1`] = `
<h1
  className="text-5xl md:text-6xl font-black bg-gradient-to-r from-pink-600 via-blue-700 to-blue-700 bg-clip-text text-transparent "
>
  We are Answer Digital.
</h1>
`;

exports[`Renders PrimaryPageText correctly when passing className 1`] = `
<h1
  className="text-5xl md:text-6xl font-black bg-gradient-to-r from-pink-600 via-blue-700 to-blue-700 bg-clip-text text-transparent pb-6"
>
  We are Answer Digital.
</h1>
`;
